src/Sarif/Converters/FxCopConverter.cs

```diff
@@ -131,13 +131,75 @@ internal Result CreateResult(FxCopLogReader.Context context)
 
             result.Locations = new List<Location> { location };
 
-            TryAddProperty(result, context.Level, "Level");
+            bool mapsDirectlyToSarifName;
+
+            result.Level = ConvertFxCopLevelToResultLevel(context.Level, out mapsDirectlyToSarifName);
+
+            if (!mapsDirectlyToSarifName)
+            {
+                // We will not recapitulate FxCop MessageLevel names (such as 
+                // "Error" and "Warning") as a property. For names that differ
+                // (such as "CriticalWarning" and "Information"), we will also 
+                // include the FxCop-specific values in the property bag.
+                TryAddProperty(result, context.Level, "Level");
```

Comment:
I like this simple rule, which is crystal clear and avoids lots of issues: if a converter perceives value in some tool-specific results data for which there is no equivalent SARIF slot, they can choose to populate the property bag with the information. If that data is named by the tool (via XML element or attribute, for example), the original tool name should be used, and the converter should preserve casing and values whatever is stored.